sdk/src/TokenRequest.cs

```diff
@@ -0,0 +1,322 @@
+        private Proto.Common.TokenProtos.TokenRequestOptions tokenRequestOptions;
+
+        public Proto.Common.TokenProtos.TokenRequestOptions GetTokenRequestOptions()
+        {
+            return tokenRequestOptions;
+        }
+
+        public TokenRequestPayload GetTokenRequestPayload()
+        {
+            return tokenRequestPayload;
+        }
+
+        /// <summary>
+        /// Create a new Builder instance for an access token request.
+        /// </summary>
+        /// <param name="resources">access token resources</param>
+        /// <returns>Builder instance</returns>
+        public static AccessBuilder accessTokenRequestBuilder(
+            params TokenRequestPayload.Types.AccessBody.Types.ResourceType[] resources)
+        {
+            return new AccessBuilder(resources);
+        }
+        
+        /// <summary>
+        /// Create a new Builder instance for an transfer token request.
+        /// </summary>
+        /// <param name="amount">lifetime amount of the token request</param>
+        /// <param name="currency">currency of the token request</param>
+        /// <returns>Builder instance</returns>
+        public static TransferBuilder transferTokenRequestBuilder(double amount, string currency)
+        {
+            return new TransferBuilder(amount, currency);
+        }
+
+        private TokenRequest(
+            TokenRequestPayload payload,
+            Proto.Common.TokenProtos.TokenRequestOptions options)
+        {
+            this.tokenRequestOptions = options;
+            this.tokenRequestPayload = payload;
+        }
+
+        /// <summary>
+        /// Creates an instance from TokenRequestPayload and TokenRequestOptions protos.
+        /// </summary>
+        /// <param name="tokenRequestPayload">TokenRequestPayload</param>
+        /// <param name="tokenRequestOptions">TokenRequestOptions</param>
+        /// <returns></returns>
+        public static TokenRequest Create(
+            TokenRequestPayload tokenRequestPayload,
+            Proto.Common.TokenProtos.TokenRequestOptions tokenRequestOptions)
+        {
+            return new TokenRequest(tokenRequestPayload, tokenRequestOptions);
+        }
+
+        public class Builder
+        {
+            protected TokenRequestPayload requestPayload;
+            protected Proto.Common.TokenProtos.TokenRequestOptions requestOptions;
+            protected string oauthState;
+            protected string csrfToken;
+
+            public Builder()
+            {
+                this.requestOptions = new Proto.Common.TokenProtos.TokenRequestOptions();
+                this.requestPayload = new TokenRequestPayload();
+            }
+
+            /// <summary>
+            /// Optional. Sets the bank ID in order to bypass the Token Bank selection UI.
+            /// </summary>
+            /// <param name="bankId">bank id</param>
+            /// <returns>builder</returns>
+            public Builder SetBankId(string bankId)
```

Comment:
Returning Builder is problematic because this means you can't `.SetBankId` and then `.AddDestinationCountry`, because `.AddDestinationCountry` is a TransferTokenBuilder method.

Reply:
Thanks for the catch. Fixed it.